### PR TITLE
Update API docs.

### PR DESCRIPTION
**Pull request by alekseysidorov**

### CODE REVIEW

File: src/architecture/services.md [hunk 1]
```diff
@@ -457,10 +457,10 @@ fn private_api_handler(&self, context: &ApiContext)
 }
 ```
 
-`public_api_handler` and `private_api_handler` provide hooks for defining
-public and private API endpoints respectively using [Iron framework][iron].
-These methods receive an API context, which allows to read information from
-the blockchain, and to translate POST requests into Exonum transactions.
+`wire_api` provide hooks for defining public and private API endpoints
+respectively using [`actix-web`][actix-web].
+These methods receive an `ServiceApiBuilder`, which allows bind
```
Comment (MartaDubno) [inline src/architecture/services.md]: Perhaps 'allows binding'?

File: src/get-started/create-service.md [hunk 2]
```diff
@@ -376,20 +364,19 @@ impl Transaction for TxTransfer {
 
 ## Implement API
 
-Next, we need to implement the node API with the help of [Iron framework][iron].
-With this aim we declare a struct which implements the `Api` trait.
-The struct will contain a channel, i.e. a connection to the blockchain node
+Next, we need to implement the node API with the help of [Actix-web framework][actix-web].
+With this aim we declare a blank struct with set of methods with the following signature:
```
Comment (MartaDubno) [inline src/get-started/create-service.md]: Many 'with' in this sentence. Perhaps we could rewrite it to something like:
'With this aim we declare a blank struct that includes a set of methods with the following signature'

Or if the meaning of 'declare a blank struct with set of methods' means using these methods, then:
'With this aim we declare a blank struct using a set of methods with the following signature'

File: src/get-started/create-service.md [hunk 3]
```diff
@@ -376,20 +364,19 @@ impl Transaction for TxTransfer {
...
+```rust
+fn my_method(state: &ServiceApiState, query: MyQueryParam) -> api::Result<MyResponseParam>
+```
+The `state` contains a channel, i.e. a connection to the blockchain node
 instance.
-Besides the channel, the API struct will contain a blockchain instance;
-it will be needed to implement
+Besides the channel, it is also contains a blockchain instance;
+this will be needed to implement
```
Comment (MartaDubno) [inline src/get-started/create-service.md]: ...a blockchain instance, which will be needed...

File: src/get-started/create-service.md [hunk 4]
```diff
@@ -440,94 +415,62 @@ We want to implement 2 read requests:
 - Return the information about a specific wallet identified by the public key.
 
 To accomplish this, we define a couple of corresponding methods in
-`CryptocurrencyApi`
-that use its `blockchain` field to read information from the blockchain
- storage.
+`CryptocurrencyApi` that use `state` to read information from the blockchain storage.
+For parsing a public key of specific wallet we define a helper structure.
```
Comment (MartaDubno) [inline src/get-started/create-service.md]: ...of a specific wallet...

File: src/get-started/create-service.md [hunk 5]
```diff
@@ -562,8 +505,7 @@ should return [a vector of hashes](../architecture/services.md#state-hash) of th
 As the wallets table is not Merkelized (a simplifying assumption discussed at the
 beginning of the tutorial), the returned value should be an empty vector, `vec![]`.
 
-The remaining method, `public_api_handler`, creates a REST `Handler` to process
-web requests to the node. We will use it to receive transactions via REST API
+The remaining method, `wire_api`, binds APIs defined by service. We will use it to receive transactions via REST API
```
Comment (MartaDubno) [inline src/get-started/create-service.md]: defined by the service

File: src/advanced/configuration-updater.md [hunk 6]
```diff
@@ -180,14 +180,14 @@ JSON object with the following fields:
...
-GET {base_path}/configs/{config_hash}
+GET {base_path}/configs?hash={config_hash}
 ```
 
 Looks up configuration (including proposals) by the hash.
 
 #### Parameters
 
-- **config_hash**: Hash  
+- **hash**: Hash=
```
Comment (vitvakatu) [inline src/advanced/configuration-updater.md]: Why did you add equal sign? I think it shouldn't be here

File: src/get-started/create-service.md [hunk 7]
```diff
@@ -376,20 +364,19 @@ impl Transaction for TxTransfer {
 
 ## Implement API
 
-Next, we need to implement the node API with the help of [Iron framework][iron].
-With this aim we declare a struct which implements the `Api` trait.
-The struct will contain a channel, i.e. a connection to the blockchain node
+Next, we need to implement the node API with the help of [Actix-web framework][actix-web].
```
Comment (vitvakatu) [inline src/get-started/create-service.md]: Do user really interacts with actix? With iron, we used its types, but do we really need to say about actix here?

File: src/get-started/create-service.md [hunk 8]
```diff
@@ -376,20 +364,19 @@ impl Transaction for TxTransfer {
...
+With this aim we declare a blank struct with set of methods with the following signature:
+```rust
+fn my_method(state: &ServiceApiState, query: MyQueryParam) -> api::Result<MyResponseParam>
+```
+The `state` contains a channel, i.e. a connection to the blockchain node
 instance.
-Besides the channel, the API struct will contain a blockchain instance;
-it will be needed to implement
+Besides the channel, it is also contains a blockchain instance;
```
Comment (vitvakatu) [inline src/get-started/create-service.md]: "is" is redundant here

File: src/architecture/services.md [hunk 9]
```diff
@@ -445,7 +445,7 @@ and push them in the queue for broadcasting, etc.
 ### REST API Initialization
 
 ```rust
-use iron::Handler;
+use exonum::api::ServiceApiBuilder;
 
 fn public_api_handler(&self, context: &ApiContext)
```
Comment (slowli) [inline src/architecture/services.md]: Please update the code sample, too. As far as I understand, `wire_api` should be mentioned here.

File: src/architecture/services.md [hunk 10]
```diff
@@ -457,10 +457,10 @@ fn private_api_handler(&self, context: &ApiContext)
 }
 ```
 
-`public_api_handler` and `private_api_handler` provide hooks for defining
-public and private API endpoints respectively using [Iron framework][iron].
-These methods receive an API context, which allows to read information from
-the blockchain, and to translate POST requests into Exonum transactions.
+`wire_api` provide hooks for defining public and private API endpoints
+respectively using [`actix-web`][actix-web].
```
Comment (slowli) [inline src/architecture/services.md]: I think this could be better:

> `wire_api` provides hooks for defining public and private API endpoints of the service.

`actix-web` seems like an implementation detail in this context.

File: src/advanced/node-management.md [hunk 11]
```diff
@@ -406,18 +409,18 @@ all messages in the event queue are processed.
 
 #### Parameters
 
-None.
+null
```
Comment (slowli) [inline src/advanced/node-management.md]: I'm not sure I can agree with this change. *Logically,* there are no parameters; passing `null` to the server is a convention.

File: src/advanced/configuration-updater.md [hunk 12]
```diff
@@ -432,7 +432,7 @@ and its signature are computed automatically.
 
 #### Parameters
 
-- **config_hash_vote_for**: Hash  
+- **hash**: Hash
```
Comment (slowli) [inline src/advanced/configuration-updater.md]: Is `hash` passed in the request body? If so, please specify, like in the "Submit Configuration Proposal" section.

File: src/get-started/create-service.md (hunk 2 above)
Comment (alekseysidorov) [reply src/get-started/create-service.md]: First variant is preferred for me.

**Comment by vitvakatu** [comment]
I suppose this PR should wait until next Exonum release?

File: src/advanced/node-management.md [hunk 13]
```diff
@@ -237,18 +237,25 @@ will not be established.
 
 #### Parameters
 
-- **ip**: PeerAddress
+- **address**: PeerAddress
```
Comment (dmitry-timofeev) [inline src/advanced/node-management.md]: I think what `PeerAddress` is requires additional clarification.

File: src/get-started/create-service.md [hunk 14]
```diff
@@ -440,94 +418,64 @@ We want to implement 2 read requests:
...
-        router.get("/v1/wallets", get_wallets, "get_wallets");
-        router.get("/v1/wallet/:pub_key", get_wallet, "get_wallet");
+impl CryptocurrencyApi {
+    fn wire(builder: &mut ServiceApiBuilder) {
+        // Binds handlers to specific routes.
+        builder
+            .public_scope()
+            .endpoint("v1/wallet", Self::get_wallet)
+            .endpoint("v1/wallets", Self::get_wallets)
+            .endpoint_mut("v1/wallets", Self::post_transaction)
```
Comment (dmitry-timofeev) [inline src/get-started/create-service.md]: Why two separate methods (`endpoint` and `endpoint_mut`)? That's the only place where they appear in the docs.

File: src/get-started/create-service.md [hunk 15]
```diff
@@ -440,94 +418,64 @@ We want to implement 2 read requests:
...
+            .endpoint_mut("v1/wallets/transfer", Self::post_transaction);
```
Comment (dmitry-timofeev) [inline src/get-started/create-service.md]: Do we bind `post_transaction` handler to several resources on purpose?

File: src/get-started/create-service.md (hunk 14 above)
Comment (alekseysidorov) [reply src/get-started/create-service.md]: Single `endpoint` have been splitted to `endpoint` and `endpoint_mut` as result of a long discussion. We need to distinguish write and read requests and thus understand where is a get request and where is a post request in HTTP backend.
I can write more detailed description about the `endpoint` and `endpoint_mut` purposes.

File: src/get-started/create-service.md (hunk 15 above)
Comment (alekseysidorov) [reply src/get-started/create-service.md]: Old code did this in the same way. Handler is same for each resource.

File: src/get-started/create-service.md (hunk 14 above)
Comment (dmitry-timofeev) [reply src/get-started/create-service.md]: Thanks, what the reader of this would benefit most? Maybe, a short explanation of the differences + a link to the API documentation on `docs.rs` (I hope it explains the differences well)?

File: src/get-started/create-service.md [hunk 16]
```diff
@@ -440,94 +418,64 @@ We want to implement 2 read requests:
...
 
 To accomplish this, we define a couple of corresponding methods in
-`CryptocurrencyApi`
-that use its `blockchain` field to read information from the blockchain
- storage.
+`CryptocurrencyApi` that use `state` to read information from the blockchain storage.
+For parsing a public key of a specific wallet we define a helper structure.
 
 ```rust
+
```
Comment (slowli) [inline src/get-started/create-service.md]: Nit: unnecessary blank line.

File: src/architecture/services.md [hunk 17]
```diff
@@ -445,22 +445,16 @@ and push them in the queue for broadcasting, etc.
...
-    None
 }
 ```
 
-`public_api_handler` and `private_api_handler` provide hooks for defining
-public and private API endpoints respectively using [Iron framework][iron].
-These methods receive an API context, which allows to read information from
-the blockchain, and to translate POST requests into Exonum transactions.
+`wire_api` provides hooks for defining public and private API endpoints of the service.
+This method receive an `ServiceApiBuilder`, which allows binding our
```
Comment (slowli) [inline src/architecture/services.md]: > This method receives a `ServiceApiBuilder`, which allows binding custom handlers to REST API.
